Panic: Post "%22<client-id>%22": unsupported protocol scheme ""

## Prerequisites

Please Check that your issue isn't already filed: [Open Issues](https://github.com/brittandeyoung/terraform-provider-awsteam/issues)

## Description

Running the awsteam provider, I get a panic

## Configuration and Steps used to Reproduce

**Expected behavior:**

For the `terraform plan` to complete ok.

**Actual behavior:**

In the below output, I have replaced my actual client secret with "x"'s.

```
$ TF_LOG=debug terraform plan
2024-02-23T10:17:22.578+1300 [INFO]  Terraform version: 1.7.4
2024-02-23T10:17:22.579+1300 [DEBUG] using github.com/hashicorp/go-tfe v1.41.0
2024-02-23T10:17:22.579+1300 [DEBUG] using github.com/hashicorp/hcl/v2 v2.19.1
2024-02-23T10:17:22.579+1300 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2024-02-23T10:17:22.579+1300 [DEBUG] using github.com/zclconf/go-cty v1.14.1
2024-02-23T10:17:22.579+1300 [INFO]  Go runtime version: go1.21.5
2024-02-23T10:17:22.579+1300 [INFO]  CLI args: []string{"terraform", "plan"}
2024-02-23T10:17:22.579+1300 [DEBUG] Attempting to open CLI config file: /home/ubuntu/.terraformrc
2024-02-23T10:17:22.579+1300 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2024-02-23T10:17:22.579+1300 [INFO]  Loading CLI configuration from /home/ubuntu/.terraform.d/credentials.tfrc.json
2024-02-23T10:17:22.579+1300 [DEBUG] checking for credentials in "/home/ubuntu/.terraform.d/plugins"
2024-02-23T10:17:22.579+1300 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2024-02-23T10:17:22.579+1300 [DEBUG] will search for provider plugins in /home/ubuntu/.terraform.d/plugins
2024-02-23T10:17:22.579+1300 [DEBUG] ignoring non-existing provider search directory /home/ubuntu/.local/share/terraform/plugins
2024-02-23T10:17:22.579+1300 [DEBUG] ignoring non-existing provider search directory /usr/local/share/terraform/plugins
2024-02-23T10:17:22.579+1300 [DEBUG] ignoring non-existing provider search directory /usr/share/terraform/plugins
2024-02-23T10:17:22.579+1300 [DEBUG] ignoring non-existing provider search directory /var/lib/snapd/desktop/terraform/plugins
2024-02-23T10:17:22.581+1300 [INFO]  CLI command args: []string{"plan"}
2024-02-23T10:17:22.585+1300 [DEBUG] New state was assigned lineage "134ff6ac-ebb2-7329-234f-f0742e3eb698"
2024-02-23T10:17:22.606+1300 [DEBUG] checking for provisioner in "."
2024-02-23T10:17:22.616+1300 [DEBUG] checking for provisioner in "/usr/bin"
2024-02-23T10:17:22.616+1300 [DEBUG] checking for provisioner in "/home/ubuntu/.terraform.d/plugins"
2024-02-23T10:17:22.617+1300 [INFO]  backend/local: starting Plan operation
2024-02-23T10:17:22.618+1300 [DEBUG] created provider logger: level=debug
2024-02-23T10:17:22.618+1300 [INFO]  provider: configuring client automatic mTLS
2024-02-23T10:17:22.644+1300 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 args=[".terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0"]
2024-02-23T10:17:22.645+1300 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243833
2024-02-23T10:17:22.645+1300 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0
2024-02-23T10:17:22.650+1300 [INFO]  provider.terraform-provider-awsteam_v0.2.0: configuring server automatic mTLS: timestamp="2024-02-23T10:17:22.650+1300"
2024-02-23T10:17:22.690+1300 [DEBUG] provider: using plugin: version=6
2024-02-23T10:17:22.690+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: plugin address: network=unix address=/tmp/plugin982899431 timestamp="2024-02-23T10:17:22.689+1300"
2024-02-23T10:17:22.716+1300 [DEBUG] No provider meta schema returned
2024-02-23T10:17:22.717+1300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-23T10:17:22.718+1300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243833
2024-02-23T10:17:22.718+1300 [DEBUG] provider: plugin exited
2024-02-23T10:17:22.719+1300 [DEBUG] Building and walking validate graph
2024-02-23T10:17:22.719+1300 [DEBUG] ProviderTransformer: "awsteam_settings.team" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/brittandeyoung/awsteam"]
2024-02-23T10:17:22.719+1300 [DEBUG] ReferenceTransformer: "awsteam_settings.team" references: []
2024-02-23T10:17:22.719+1300 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/brittandeyoung/awsteam\"]" references: []
2024-02-23T10:17:22.720+1300 [DEBUG] Starting graph walk: walkValidate
2024-02-23T10:17:22.720+1300 [DEBUG] created provider logger: level=debug
2024-02-23T10:17:22.720+1300 [INFO]  provider: configuring client automatic mTLS
2024-02-23T10:17:22.729+1300 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 args=[".terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0"]
2024-02-23T10:17:22.729+1300 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243841
2024-02-23T10:17:22.729+1300 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0
2024-02-23T10:17:22.734+1300 [INFO]  provider.terraform-provider-awsteam_v0.2.0: configuring server automatic mTLS: timestamp="2024-02-23T10:17:22.734+1300"
2024-02-23T10:17:22.768+1300 [DEBUG] provider: using plugin: version=6
2024-02-23T10:17:22.768+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: plugin address: address=/tmp/plugin1911962344 network=unix timestamp="2024-02-23T10:17:22.768+1300"
2024-02-23T10:17:22.796+1300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-23T10:17:22.798+1300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243841
2024-02-23T10:17:22.798+1300 [DEBUG] provider: plugin exited
2024-02-23T10:17:22.798+1300 [INFO]  backend/local: plan calling Plan
2024-02-23T10:17:22.798+1300 [DEBUG] Building and walking plan graph for NormalMode
2024-02-23T10:17:22.798+1300 [DEBUG] ProviderTransformer: "awsteam_settings.team (expand)" (*terraform.nodeExpandPlannableResource) needs provider["registry.terraform.io/brittandeyoung/awsteam"]
2024-02-23T10:17:22.798+1300 [DEBUG] ReferenceTransformer: "awsteam_settings.team (expand)" references: []
2024-02-23T10:17:22.799+1300 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/brittandeyoung/awsteam\"]" references: []
2024-02-23T10:17:22.799+1300 [DEBUG] Starting graph walk: walkPlan
2024-02-23T10:17:22.799+1300 [DEBUG] created provider logger: level=debug
2024-02-23T10:17:22.799+1300 [INFO]  provider: configuring client automatic mTLS
2024-02-23T10:17:22.816+1300 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 args=[".terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0"]
2024-02-23T10:17:22.816+1300 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243849
2024-02-23T10:17:22.816+1300 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0
2024-02-23T10:17:22.822+1300 [INFO]  provider.terraform-provider-awsteam_v0.2.0: configuring server automatic mTLS: timestamp="2024-02-23T10:17:22.822+1300"
2024-02-23T10:17:22.858+1300 [DEBUG] provider: using plugin: version=6
2024-02-23T10:17:22.858+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: plugin address: address=/tmp/plugin1544466970 network=unix timestamp="2024-02-23T10:17:22.857+1300"
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: panic: Post "%227h31pg0o0fokuk5gb97d98j2bi%22": unsupported protocol scheme ""
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: goroutine 49 [running]:
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/brittandeyoung/terraform-provider-awsteam/internal/sdk/awsteam.(*Config).Build(0xc0000ba660)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/brittandeyoung/terraform-provider-awsteam/internal/sdk/awsteam/config.go:62 +0x434
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/brittandeyoung/terraform-provider-awsteam/internal/provider.(*AWSTEAMProvider).Configure(0x0?, {0xd191e8, 0xc00009c2d0}, {{0xc00040e448, 0x5}, {{{0xd1d448, 0xc00009cbd0}, {0xb1f500, 0xc00009cae0}}, {0xd1ea00, ...}}}, ...)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/brittandeyoung/terraform-provider-awsteam/internal/provider/provider.go:105 +0x3c5
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc000147600, {0xd191e8, 0xc00009c2d0}, 0xc0000a61c0, 0xc0000a6140)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_configureprovider.go:18 +0x10f
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0xc000147600, {0xd191e8?, 0xc00047b9e0?}, 0xc00047b9b0?)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_configureprovider.go:39 +0x30f
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0xc00018d180, {0xd191e8?, 0xc00047b1d0?}, 0xc000430cc0)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:568 +0x2d3
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0xbdf040?, 0xc00018d180}, {0xd191e8, 0xc00047b1d0}, 0xc000402c00, 0x0)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:449 +0x170
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001d01e0, {0xd191e8, 0xc00047b140}, {0xd1d778, 0xc000394000}, 0xc000484000, 0xc0002eb170, 0x11ae7d0, 0x0)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	google.golang.org/grpc@v1.60.0/server.go:1372 +0xe49
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: google.golang.org/grpc.(*Server).handleStream(0xc0001d01e0, {0xd1d778, 0xc000394000}, 0xc000484000)
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	google.golang.org/grpc@v1.60.0/server.go:1783 +0x1031
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	google.golang.org/grpc@v1.60.0/server.go:1016 +0x68
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: created by google.golang.org/grpc.(*Server).serveStreams.func2
2024-02-23T10:17:22.892+1300 [DEBUG] provider.terraform-provider-awsteam_v0.2.0: 	google.golang.org/grpc@v1.60.0/server.go:1027 +0x12e
2024-02-23T10:17:22.893+1300 [ERROR] plugin6.(*GRPCProvider).ConfigureProvider: error="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-23T10:17:22.893+1300 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/brittandeyoung/awsteam/0.2.0/linux_amd64/terraform-provider-awsteam_v0.2.0 pid=2243849 error="exit status 2"
2024-02-23T10:17:22.893+1300 [ERROR] vertex "provider[\"registry.terraform.io/brittandeyoung/awsteam\"]" error: Plugin did not respond
2024-02-23T10:17:22.893+1300 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2024-02-23T10:17:22.894+1300 [INFO]  backend/local: plan operation completed

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-awsteam_v0.2.0 plugin:

panic: Post "%22xxxxxxxxxxxxxxxxxxxxxxxx%22": unsupported protocol scheme ""

goroutine 49 [running]:
github.com/brittandeyoung/terraform-provider-awsteam/internal/sdk/awsteam.(*Config).Build(0xc0000ba660)
	github.com/brittandeyoung/terraform-provider-awsteam/internal/sdk/awsteam/config.go:62 +0x434
github.com/brittandeyoung/terraform-provider-awsteam/internal/provider.(*AWSTEAMProvider).Configure(0x0?, {0xd191e8, 0xc00009c2d0}, {{0xc00040e448, 0x5}, {{{0xd1d448, 0xc00009cbd0}, {0xb1f500, 0xc00009cae0}}, {0xd1ea00, ...}}}, ...)
	github.com/brittandeyoung/terraform-provider-awsteam/internal/provider/provider.go:105 +0x3c5
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ConfigureProvider(0xc000147600, {0xd191e8, 0xc00009c2d0}, 0xc0000a61c0, 0xc0000a6140)
	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/fwserver/server_configureprovider.go:18 +0x10f
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ConfigureProvider(0xc000147600, {0xd191e8?, 0xc00047b9e0?}, 0xc00047b9b0?)
	github.com/hashicorp/terraform-plugin-framework@v1.4.2/internal/proto6server/server_configureprovider.go:39 +0x30f
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0xc00018d180, {0xd191e8?, 0xc00047b1d0?}, 0xc000430cc0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:568 +0x2d3
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0xbdf040?, 0xc00018d180}, {0xd191e8, 0xc00047b1d0}, 0xc000402c00, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:449 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001d01e0, {0xd191e8, 0xc00047b140}, {0xd1d778, 0xc000394000}, 0xc000484000, 0xc0002eb170, 0x11ae7d0, 0x0)
	google.golang.org/grpc@v1.60.0/server.go:1372 +0xe49
google.golang.org/grpc.(*Server).handleStream(0xc0001d01e0, {0xd1d778, 0xc000394000}, 0xc000484000)
	google.golang.org/grpc@v1.60.0/server.go:1783 +0x1031
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.60.0/server.go:1016 +0x68
created by google.golang.org/grpc.(*Server).serveStreams.func2
	google.golang.org/grpc@v1.60.0/server.go:1027 +0x12e

Error: The terraform-provider-awsteam_v0.2.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2024-02-23T10:17:22.895+1300 [DEBUG] provider: plugin exited
```

**Reproduces how often:**

Always. This behaviour occurs when running with terraform 1.7.4 from the CLI, but also in 1.2.5 in terraform cloud (those are the only two versions I have tried)

### Provider Version

 $ terraform version
Terraform v1.7.4
on linux_amd64
+ provider registry.terraform.io/brittandeyoung/awsteam v0.2.0

### Additional Information

Test case: (Secrets replaced with "x"'s)

```hcl
terraform {
    required_providers {
        awsteam = {
            source  = "brittandeyoung/awsteam"
            version = "0.2.0"
        }
    }
}


provider "awsteam" {
  token_endpoint = "https://xxxxxxxxxxxxx-main.auth.ap-southeast-2.amazoncognito.com/oauth2/token"
  graph_endpoint = "https://xxxxxxxxxxxxxxxxxxxxxxxx.appsync-api.ap-southeast-2.amazonaws.com/graphql"
  client_id      = "xxxxxxxxxxxxxxxxxxx"
  client_secret  = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
}

resource "awsteam_settings" "team" {
    duration                  = 1 # Max length of a request (hours)
    expiry                    = 3 # Hours before an unapproved request expires
    team_admin_group          = "Role: RoleNameHere"
    team_auditor_group        = "Role: RoleNameHere"
    approval                  = true # We can require approval on workflows, if we want to
    comments                  = true # comments are required
    ticket_no                 = true # ticket numbers are required
    sns_notifications_enabled = false # TODO: implement this
}

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,8 @@ This project adheres to [Semantic Versioning](http://semver.org/) and [Keep a Ch
 ### Changes
 
 ### Fixes
+* Provider: Corrected the provider schema attribute mapping to configuration fields for `client_secret`, `graph_endpoint`, and `token_endpoint` [24](https://github.com/brittandeyoung/terraform-provider-awsteam/issues/24)
+
 
 ### Breaks
 

--- a/internal/provider/provider.go
+++ b/internal/provider/provider.go
@@ -87,9 +87,9 @@ func (p *AWSTEAMProvider) Configure(ctx context.Context, req provider.ConfigureR
 	}
 
 	clientId := fieldOrEnvVar(data.ClientId, "client_id", envvar.AWSTEAMClientId, resp)
-	clientSecret := fieldOrEnvVar(data.ClientId, "client_secret", envvar.AWSTEAMClientSecret, resp)
-	graphEndpoint := fieldOrEnvVar(data.ClientId, "graph_endpoint", envvar.AWSTEAMGraphEndpoint, resp)
-	TokenEndpoint := fieldOrEnvVar(data.ClientId, "token_endpoint", envvar.AWSTEAMTokenEndpoint, resp)
+	clientSecret := fieldOrEnvVar(data.ClientSecret, "client_secret", envvar.AWSTEAMClientSecret, resp)
+	graphEndpoint := fieldOrEnvVar(data.GraphEndpoint, "graph_endpoint", envvar.AWSTEAMGraphEndpoint, resp)
+	TokenEndpoint := fieldOrEnvVar(data.TokenEndpoint, "token_endpoint", envvar.AWSTEAMTokenEndpoint, resp)
 
 	if resp.Diagnostics.HasError() {
 		return
@@ -102,7 +102,7 @@ func (p *AWSTEAMProvider) Configure(ctx context.Context, req provider.ConfigureR
 		TokenEndpoint: TokenEndpoint,
 	}
 
-	config.Build()
+	config.Build(ctx)
 
 	meta := config.NewClient(ctx)
 
@@ -142,7 +142,7 @@ func fieldOrEnvVar(field basetypes.StringValue, fieldName string, envvarName str
 			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Providing a value for %s is required. This can also be handled by setting the %s environment variable.", fieldName, envvarName))
 		}
 	} else {
-		value = field.String()
+		value = field.ValueString()
 	}
 	return value
 }

--- a/internal/sdk/awsteam/config.go
+++ b/internal/sdk/awsteam/config.go
@@ -8,6 +8,7 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hasura/go-graphql-client"
 	"golang.org/x/oauth2"
 )
@@ -43,11 +44,12 @@ type Config struct {
 	TokenEndpoint string
 }
 
-func (config *Config) Build() {
+func (config *Config) Build(ctx context.Context) {
 	// Configure the AWS TEAM client
 	// First we need to get a token from the oath endpoint
 	authPayload := strings.NewReader(fmt.Sprintf(`grant_type=client_credentials&scope=api%%2Fadmin&client_id=%s&client_secret=%s`, config.ClientId, config.ClientSecret))
 
+	tflog.Debug(ctx, "Preparing token request", map[string]interface{}{"token_endpoint": config.TokenEndpoint, "graph_endpoint": config.GraphEndpoint, "client_id": config.ClientId})
 	authClient := &http.Client{}
 	authReq, err := http.NewRequest("POST", config.TokenEndpoint, authPayload)
 
@@ -59,6 +61,7 @@ func (config *Config) Build() {
 	res, err := authClient.Do(authReq)
 
 	if err != nil {
+		tflog.Error(ctx, "Data provided is invalid. Unable to build request for token endpoint.")
 		panic(err)
 	}
 
@@ -67,6 +70,7 @@ func (config *Config) Build() {
 	body, err := io.ReadAll(res.Body)
 
 	if err != nil {
+		tflog.Error(ctx, "Failed to receive token from endpoint.")
 		panic(err)
 	}
 
@@ -75,6 +79,7 @@ func (config *Config) Build() {
 	err = json.Unmarshal(body, token)
 
 	if err != nil {
+		tflog.Error(ctx, "Invalid JSON in response. Unmarshalling failed.")
 		panic(err)
 	}
 

